llvm/utils/repo/docker/scripts/checkout_pstore.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+#===- llvm/utils/repo/docker/scripts/checkout_pstore.sh -------------------===//
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+#===-----------------------------------------------------------------------===//
+
+set -e
+
+function show_usage() {
+  cat << EOF
+Usage: checkout_pstore.sh [options]
+
+Checkout git sources into /tmp/pstore-build/src. Used inside a docker container.
+
+Available options:
+  -h|--help           show this help message
+  -b|--branch         git branch to checkout, i.e. 'trunk',
+                      'branches/release_40'
+                      (default: 'trunk')
+  -l|--local          use local repository
+  -w|--workspace      checkout and build workspace (internal)
+EOF
+}
+
+function clone_project() {
+  local PROJECT="$1"
+  local DIRECTORY="$2"
+  local DESTINATION="$3"
+  local SOURCE=$BRANCH
+
+  # Check if remote branch exists.
+  set +e
+  git ls-remote --heads --exit-code $GIT_REPOSITORY/$PROJECT $SOURCE
+  if [ "$?" == "2" ]; then
+    echo "Branch '$SOURCE' does not exist. Using 'master'."
+    SOURCE="master"
+  fi
+  set -e
+  echo "Checking out '$SOURCE/$PROJECT' into '$DIRECTORY/$DESTINATION'"
+  git clone --branch $SOURCE $GIT_REPOSITORY/$PROJECT $DESTINATION
```

Comment:
Could we limit the amount of history that gets checked out when the project is cloned? On my ropey connection here, it took four attempts before I could get past this step!

Reply:
Very good point. Adding '--depth 1' to the git clone command. Gets only the last revision.